apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -572,10 +572,18 @@ def _write_dataframe(
     df.set_index(SOMA_JOINID, inplace=True)
 
     # Categoricals are not yet well supported, so we must flatten
+    null_fields = set()
     for k in df:
         if df[k].dtype == "category":
             df[k] = df[k].astype(df[k].cat.categories.dtype)
+        elif df[k].isnull().all():
+            df[k] = pa.nulls(df.shape[0], pa.infer_type(df[k]))
+            null_fields.add(k)
```

Comment:
Can these two both be true? i.e., a category might be all-nulls?

(also, should this be if *any* entry is null?)

Reply:
Yes and yes, thanks for catching!